source/MaterialXFormat/XmlIo.cpp

```diff
@@ -332,10 +332,12 @@ void readFromXmlFile(DocumentPtr doc, FilePath filename, FileSearchPath searchPa
     documentFromXml(doc, xmlDoc, searchPath, readOptions);
 }
 
-void readFromXmlString(DocumentPtr doc, const string& str, const XmlReadOptions* readOptions)
+void readFromXmlString(DocumentPtr doc, const string& str, FileSearchPath searchPath, const XmlReadOptions* readOptions)
 {
+    searchPath.append(getEnvironmentPath());
```

Comment:
Wouldn't it be better to do this in fromStream and fromBuffer? Otherwise, these functions won't take the environment path into account when being called directly. It has to be removed in fromString again then, to not append it twice.

Reply:
By the way, I noticed that since `readFromXmlFile` is used to read includes by default, the environment variable is appended to the search path for every include. This will only affect performance in case a file isn't found at all, so it does probably not matter much.